scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -157,35 +160,50 @@ func getPipelineKey(resourceName string, isModel bool) string {
 	}
 }
 
-func (km *KafkaManager) loadOrStorePipeline(resourceName string, isModel bool) (*Pipeline, error) {
-	logger := km.logger.WithField("func", "loadOrStorePipeline")
+func loadPipeline(resourceName string, isModel bool, pipelines *sync.Map) (*Pipeline, error) {
 	key := getPipelineKey(resourceName, isModel)
-	if val, ok := km.pipelines.Load(key); ok {
-		val.(*Pipeline).wg.Wait()
+	if val, ok := pipelines.Load(key); ok {
 		return val.(*Pipeline), nil
-	} else {
-		pipeline, err := km.createPipeline(resourceName, isModel)
-		if err != nil {
-			return nil, err
-		}
-		pipeline.wg.Add(1) // wait set to allow consumer to say when started
+	}
+	return nil, fmt.Errorf("pipeline for resource %s not found", resourceName)
+}
 
-		val, loaded := km.pipelines.LoadOrStore(key, pipeline)
-		if loaded { // we can still have a race condition where multiple "create" are happening, we have to store the first one
-			pipeline = val.(*Pipeline)
-		} else {
-			go func() {
-				err := km.consume(pipeline)
-				if err != nil {
-					km.logger.WithError(err).Errorf("Failed running consumer for resource %s", resourceName)
-				}
-			}()
-		}
+func (km *KafkaManager) LoadPipeline(resourceName string, isModel bool) (*Pipeline, error) {
+	km.mu.RLock()
+	defer km.mu.RUnlock()
+	return loadPipeline(resourceName, isModel, &km.pipelines)
+}
+
+func (km *KafkaManager) StorePipeline(resourceName string, isModel bool) error {
```

Comment:
This is much more than just "Store", because it essentially does "LoadOrCreateConsumersAndStore". Given that we can wait on the Consumers via the wg, could this be turned into one function that uses the standard `LoadOrStore` operation, plus the logic for creating the consumers and them subscribing to the right topics?